noxfile.py

```diff
@@ -184,34 +184,48 @@ def default(session):
         *session.posargs,
     )
 
-    # XXX Work around Kokoro image's older pip, which borks the OT install.
-    session.run("pip", "install", "--upgrade", "pip")
-    session.install("-e", ".[tracing]", "-c", constraints_path)
-    # XXX: Dump installed versions to debug OT issue
-    session.run("pip", "list")
 
-    # Run py.test against the unit tests with OpenTelemetry.
+@nox.session(python=UNIT_TEST_PYTHON_VERSIONS)
+@nox.parametrize(
+    "protobuf_implementation",
+    ["python", "upb", "cpp"],
```

Comment:
@parthea Just curious to understand what does this `protobuf_implementation` mean and why do we have 3 options `["python", "upb", "cpp"]`?

Reply:
This is a feature of `protobuf` where they offer [three separate implementations](https://github.com/protocolbuffers/protobuf/tree/main/python#implementation-backends) of the `Python Protobuf` API. There is an expectation that we should support all 3 implementations in the Cloud Python SDK.

`upb` and `python ` are actively supported in the latest `Protobuf 5.x` while `cpp` is only available in `Protobuf 3.x`